src/main/java/hudson/plugins/git/util/BuildData.java

```diff
@@ -243,4 +243,22 @@ public String toString() {
                 ",buildsByBranchName="+buildsByBranchName+
                 ",lastBuild="+lastBuild+"]";
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
> Doesn't adding an equals method without a hashCode method mean that this object will violate the hashCode contract?

It does. The current calling code is searching in a `List` so it does not matter.

For purposes of this PR I am simply trying to restore what was lost. If you think #372 was mistaken to begin with, then some further development is needed, either here or in a follow-up.